ghost.py

```diff
@@ -375,6 +388,63 @@ def delete(self, key_name):
         return result.rowcount > 0
 
 
+class ConsulStorage(object):
+    def __init__(self,
+                 db_path='http://127.0.0.1:8500',
+                 directory='ghost',
+                 verify=True,
+                 client_cert=None,
+                 auth=None):
+        if not REQUESTS_EXISTS:
+            raise ImportError('Requests must be installed first')
+        self._url = urljoin(db_path, 'v1/kv/{0}/'.format(directory))
+        self._session = requests.Session()
+        self._session.verify = verify
+        self._session.cert = client_cert
+        self._session.auth = auth
+
+    def _key_url(self, key):
+        return urljoin(self._url, key)
+
+    def _consul_request(self, method, url, *args, **kwargs):
+        handler = getattr(self._session, method.lower())
+        response = handler(url, *args, **kwargs)
+        if response.status_code == 404:
+            return None
+        if response.status_code >= 400:
+            raise GhostError('{0} {1} returned {2}: {3}'.format(
+                             method, url, response.status_code,
+                             response.content))
+        return response.json()
+
+    def init(self):
+        pass
+
+    def put(self, key):
+        """
+        `key` is the dictionary representing the key
+        """
+        return self._consul_request(
+            'PUT', self._key_url(key['name']), json=key)
+
+    def list(self):
```

Comment:
I think we need something else here. The current way the storage returns keys is actually returns a list of dicts containing all keys with all of their data.
Not saying that it's good that we do that, but for consistency sake, until decided otherwise, I think this should be done:


```python
def list(self):
    all_keys = self._request('GET', self._url + '?keys')
    key_list = []
    for key in all_keys:
        key_list.append(self.get(os.path.basename(key)))
    return key_list
```

or 

```python
return [self.get(os.path.basename(key)) for key in self._request('GET', self._url + '?keys')]
```